jetty-client/src/main/java/org/eclipse/jetty/client/ResponseNotifier.java

```diff
@@ -41,8 +41,9 @@
 
     public void notifyBegin(List<Response.ResponseListener> listeners, Response response)
     {
-        for (Response.ResponseListener listener : listeners)
+        for (int i = 0; i < listeners.size(); i++)
         {
+            Response.ResponseListener listener = listeners.get(i);
```

Comment:
I don't think we need to do this - the List iterator is supposed to be simple and in this case the JIT should eliminate its allocation.

Reply:
You're right, Zulu 11.0.7 seems to be incapable of figuring out that these iterators can be stack-allocated while Zulu 1.8.0_242 does figure it out, so it seems to be a JVM problem. I'll revert this change and investigate that separately.